### ISSUE TITLE
`declare -g` is not available on the standard Mac OS X version of bash

### ISSUE THREAD

**Issue by TimMoore** [postmodern/ruby-install#219]
I just updated to the HEAD of `ruby-install`, and I now get errors when I run it:

```
$ ruby-install
/usr/local/bin/../share/ruby-install/ruby-install.sh: line 5: declare: -g: invalid option
declare: usage: declare [-afFirtx] [-p] [name[=value] ...]
/usr/local/bin/../share/ruby-install/ruby-install.sh: line 6: declare: -g: invalid option
declare: usage: declare [-afFirtx] [-p] [name[=value] ...]
/usr/local/bin/../share/ruby-install/ruby-install.sh: line 8: /versions.sh: No such file or directory
/usr/local/bin/../share/ruby-install/ruby-install.sh: line 10: declare: -g: invalid option
declare: usage: declare [-afFirtx] [-p] [name[=value] ...]
/usr/local/bin/../share/ruby-install/ruby-install.sh: line 11: declare: -g: invalid option
declare: usage: declare [-afFirtx] [-p] [name[=value] ...]
/usr/local/bin/../share/ruby-install/ruby-install.sh: line 12: declare: -g: invalid option
declare: usage: declare [-afFirtx] [-p] [name[=value] ...]
/usr/local/bin/../share/ruby-install/ruby-install.sh: line 13: declare: -g: invalid option
declare: usage: declare [-afFirtx] [-p] [name[=value] ...]
/usr/local/bin/../share/ruby-install/ruby-install.sh: line 14: declare: -g: invalid option
declare: usage: declare [-afFirtx] [-p] [name[=value] ...]
/usr/local/bin/../share/ruby-install/ruby-install.sh: line 17: declare: -g: invalid option
declare: usage: declare [-afFirtx] [-p] [name[=value] ...]
/usr/local/bin/../share/ruby-install/ruby-install.sh: line 20: declare: -g: invalid option
declare: usage: declare [-afFirtx] [-p] [name[=value] ...]
Latest ruby versions:
  ruby:
sed: /ruby/stable.txt: No such file or directory
```

It looks like this was just introduced in e9a18f49bdd266149bffa38713d7f4eb75f73004, but unfortunately doesn't work on the stock OS X version of bash:

```
$ help declare
declare: declare [-afFirtx] [-p] [name[=value] ...]
    Declare variables and/or give them attributes.  If no NAMEs are
    given, then display the values of variables instead.  The -p option
    will display the attributes and values of each NAME.
    
    The flags are:
    
      -a	to make NAMEs arrays (if supported)
      -f	to select from among function names only
      -F	to display function names (and line number and source file name if
    	debugging) without definitions
      -i	to make NAMEs have the `integer' attribute
      -r	to make NAMEs readonly
      -t	to make NAMEs have the `trace' attribute
      -x	to make NAMEs export
    
    Variables with the integer attribute have arithmetic evaluation (see
    `let') done when the variable is assigned to.
    
    When displaying values of variables, -f displays a function's name
    and definition.  The -F option restricts the display to function
    name only.
    
    Using `+' instead of `-' turns off the given attribute instead.  When
    used in a function, makes NAMEs local, as with the `local' command.
$ bash --version
GNU bash, version 3.2.57(1)-release (x86_64-apple-darwin14)
Copyright (C) 2007 Free Software Foundation, Inc.
```

**Comment by rapenchukd** [comment]
declare -g is also not available on any EL6 or EL5 platforms. This just broke our whole infrastructure!

**Comment by havenwood** [comment]
@rapenchukd It was tested in the sense that the tests were run locally and by continuous integration. As well as manual checks on several GNU/Linux distros, OS X, FreeBSD and OpenBSD with both bash and zsh. It wasn't tested on dated bash, where it unfortunately breaks.

@TimMoore Thanks for the report! Closing as the issue is fixed by fc290fbd859a0ffab1d11a061a6e56e15ddc1fde.
